### PR TITLE
Feat/login modal

### PR DESCRIPTION
**Pull request by y-golde**
adding login modal, general modal with header option and login store-localStorage funcitonality - what's left is to connect it to /auth

### CODE REVIEW

**Comment by y-golde** [comment]
maybe we'll add a logout option down the road

File: Client/src/App.svelte [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+	import LoginModal from './Components/App/LoginModal/LoginModal.svelte';
+
+	let userName: string = '';
+	const unsubscribe = user.subscribe((value) => {
+		userName = value;
+	});
 
 	const headlineText = 'ishtaloo.io';
+	let showLoginModal = false;
+	$: showLoginModal = shouldPropLogin(userName);
```
Comment (y-golde) [inline Client/src/App.svelte:15]: move up

File: Client/src/Components/App/LoginModal/LoginModal.svelte [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	}
+</script>
+
+<Modal show="{show}">
+	<ModalHeader slot="header" text="{modalHeaderText}" />
+	<div slot="content" class="text-2xl">
+		<p class="p-3">Hello, please provide a nickname:</p>
+		<div class="flex flex-wrap p-3 gap-2 items-center justify-center">
+			<Textfield bind:value="{userName}" placeholder="{placeHolderText}" />
+			<Button disabled="{disabled}" handleClick="{() => handleLoginClick(userName)}" />
```
Comment (y-golde) [inline Client/src/Components/App/LoginModal/LoginModal.svelte]: add text, mayne something like ✔, what do you think? @dstr951

File: Client/src/Store/user.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { writable } from 'svelte/store';
+
+const user = writable('');
```
Comment (y-golde) [inline Client/src/Store/user.ts]: type it (maybe - see docs)

**Review by y-golde — COMMENTED** [review]
@dstr951, I've pointed out a few minor mistakes

File: Client/src/Components/App/LoginModal/LoginModal.svelte (hunk 2 above)
Comment (dstr951) [reply Client/src/Components/App/LoginModal/LoginModal.svelte]: I think that the login should be as simple as possible, like any io game
just type your name and thats it' you can start

File: Client/src/App.svelte (hunk 1 above)
Comment (dstr951) [reply Client/src/App.svelte:15]: move up what?

File: Client/src/Utils/User/useUser.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	const existingUser = localStorage.getItem(USER_CONTEXT_NAME);
+	if (existingUser && !userName) {
+		setUser(existingUser);
+	}
+
+	return !Boolean(userName || existingUser);
+};
+
+export const setUser = (userName: string) => {
+	user.set(userName);
```
Comment (dstr951) [inline Client/src/Utils/User/useUser.ts:15]: Well, I didn't upload it yet, but in my local version user has a name and an Id, will you store the id? or not? if not then that's ok
just making sure

File: Client/src/Store/user.ts (hunk 3 above)
Comment (dstr951) [reply Client/src/Store/user.ts]: type it? I didn't understand
But yeah I think it's a good use of stores, we just to be careful with the token expiary thingi

**Review by dstr951 — APPROVED** [review]
Basically I agree with everything, I just asked some questions

File: Client/src/Utils/User/useUser.ts (hunk 4 above)
Comment (y-golde) [reply Client/src/Utils/User/useUser.ts:15]: Sure, we'll connect it in a separate pr

File: Client/src/Store/user.ts (hunk 3 above)
Comment (y-golde) [reply Client/src/Store/user.ts]: add a type!

**Pull request merged**